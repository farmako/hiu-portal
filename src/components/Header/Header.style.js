import React from 'react';
import styled from 'styled-components';
import Button from '@material-ui/core/Button';
import Box from '@material-ui/core/Box';


export const LogoutButton = styled(({ ...rest }) => <Button {...rest} variant="contained" color="primary" />)`
    margin-right: 1em;
`;

export const Logo = styled.a`
    color: #fff;
    background-color: #dedede;
    margin-left: 1em;
    text-transform: uppercase;
    text-decoration: none;
    justify-content: center;
`;


export const Header = styled(({ ...rest }) => <Box {...rest} boxShadow={3} />)`
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    position: fixed;
    top: 0;
    left: 0;
    padding: 1em 0;
    width: 100%;
    z-index: 50;
    background: #fff;
`;
