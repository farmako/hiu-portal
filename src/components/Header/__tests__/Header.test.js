import React from 'react';
import { shallow } from 'enzyme';
import Button from '@material-ui/core/Button';
import Header from '..';

describe('Header', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Header />);
    localStorage.removeItem.mockReset();
  });

  describe('HTML Structure', () => {
    it('should render properly', () => {
      expect(wrapper.debug()).toMatchSnapshot();
    });
  });

  describe('Events', () => {
    describe('logout()', () => {
      it('should call removeItem() with auth-token', () => {
        wrapper.find(Button).simulate('click');
        expect(localStorage.removeItem).toHaveBeenCalledWith('auth-token');
      });

      it('should reload the window', () => {
        wrapper.find(Button).simulate('click');
        expect(window.location.reload).toHaveBeenCalled();
      });
    });
  });
});
