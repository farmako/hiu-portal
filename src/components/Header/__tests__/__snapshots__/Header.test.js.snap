// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header HTML Structure should render properly 1`] = `
"<Fragment>
  <WithStyles(ForwardRef(AppBar)) color=\\"textPrimary\\">
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(Typography)) variant=\\"h6\\" style={{...}}>
        <WithStyles(ForwardRef(Link)) href=\\"/\\">
          LOGO
        </WithStyles(ForwardRef(Link))>
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Button)) color=\\"primary\\" variant=\\"contained\\" onClick={[Function: logout]}>
        Logout
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</Fragment>"
`;
